rest/route/middleware.go

```diff
@@ -614,106 +614,130 @@ func RequiresSuperUserPermission(permission string, level evergreen.PermissionLe
 	return gimlet.RequiresPermission(opts)
 }
 
-func urlVarsToProjectScopes(r *http.Request) ([]string, int, error) {
-	var err error
-	vars := gimlet.GetVars(r)
-	query := r.URL.Query()
-	resourceType := strings.ToUpper(util.CoalesceStrings(query["resource_type"], vars["resource_type"]))
-	if resourceType != "" {
-		switch resourceType {
-		case event.EventResourceTypeProject:
-			vars["project_id"] = vars["resource_id"]
-		case event.ResourceTypeTask:
-			vars["task_id"] = vars["resource_id"]
-		}
-	}
+func GetProjectIdForProjectScopes(ctx context.Context, paramsMap map[string]string) (string, int, error) {
```

Comment:
public functions should have a function comment describing the input and output

Reply:
Also I know we called this "ProjectScopes" before but i don't know that that makes sense, since this isn't what our definition of "scopes" is now that we have a scopes collection. Perhaps we could rename this to something like GetProjectIdFromParams or something.